src/sync/deployment.test.ts

```diff
@@ -30,6 +37,221 @@ describe("sync/deployments", () => {
+				clone.node.updatedAt = `2023-01-0${idx + 1}T10:00:00Z`;
+				clone.node.databaseId = `dbid-${idx + 1}`;
+				clone.node.commitOid = `SHA${idx + 1}`;
+				clone.node.statuses.nodes.forEach(n => {
+					n.createdAt = clone.node.createdAt;
+					n.updatedAt = clone.node.updatedAt;
+				});
+				return clone;
+			});
+		};
+
+		const nockFetchingDeploymentgPagesGraphQL = (cursor, deployments) => {
+			githubNock.post("/graphql", { query: getDeploymentsQuery, variables: { owner: repoSyncState.repoOwner, repo: repoSyncState.repoName, per_page: PAGE_SIZE__TWO_ITEMS, cursor } })
+				.query(true).reply(200, { "data": { "repository": { "deployments": { "edges": deployments } } } });
+		};
+
+		const nockDeploymentListingApi = (deployments, repeatTimes) => {
+			"*".repeat(repeatTimes).split("").forEach(() => {
+				githubNock.get(`/repos/test-repo-owner/test-repo-name/deployments?environment=prod&per_page=10`)
+					.reply(200, deployments.map((item, idx) => ({
+						"id": item.node.databaseId,
+						"sha": item.node.commitOid,
+						"ref": "random",
+						"task": `task for deployment ${idx + 1}`,
+						"payload": {},
+						"original_environment": item.node.environment,
+						"environment": item.node.environment,
+						"description": `description for deployment ${idx + 1}`,
+						"creator": { "login": "test-repo-owner", "id": 1, "type": "User" },
+						"created_at": item.node.createdAt,
+						"updated_at": item.node.createdAt,
+						"statuses_url": "random",
+						"repository_url": "random",
+						"transient_environment": false,
+						"production_environment": true
+					})));
+			});
+		};
+
+		const nockDeploymentStatusApi = (deployments, repeatTimes) => {
+			"*".repeat(repeatTimes).split("").forEach(() => {
+				deployments.forEach((item, idx) => {
+					githubNock.get(`/repos/test-repo-owner/test-repo-name/deployments/${item.node.databaseId}/statuses?per_page=100`)
+						.reply(200, [
+							{ "id": idx * 1000 + 1, "state": "inactive", "description": "random", "environment": item.node.environment },
+							{ "id": idx * 1000 + 2, "state": "success", "description": "random", "environment": item.node.environment }
+						]);
+				});
+			});
+		};
+
+		const nockDeploymentCommitGetApi = (deployments, repeatTimes) => {
+			"*".repeat(repeatTimes).split("").forEach(() => {
+				deployments.forEach(item => {
+					githubNock.get(`/repos/test-repo-owner/test-repo-name/commits/${item.node.commitOid}`)
+						.reply(200, {
+							commit: {
+								author: { name: "random", email: "random", date: new Date() },
+								message: `commit message [JIRA-${item._seq}]`
+							}, html_url: "random"
+						});
+				});
+			});
+		};
+
+		const expectDeploymentEntryInDB = async (deployments) => {
+			for (const deployment of deployments) {
+				const key = createHashWithoutSharedSecret(`ghurl_${gitHubCloudConfig.baseUrl}_repo_${deployment.node.repository.id}_env_${deployment.node.environment}`);
```

Comment:
To tighten this up a little in terms of readability can we throw in some destructuring at the start

```
const {
    repository: { id: repoId },
    environment,
    commitOid
  } = deployment.node;
```

Reply:
fixed (to my best effort 🤷 )